### PR TITLE
Add React Hooks for customization (part 12)

### PR DESCRIPTION
**Pull request by compulim**
> Related to #2539.

Please merge #2540, #2541, #2542, #2543, #2544, #2547, #2548, #2549, #2550, #2551, #2552 before reviewing this one.

## Changelog Entry

-  Resolves [#2539](https://github.com/Microsoft/BotFramework-WebChat/issues/2539), added React hooks for customization, by [@compulim](https://github.com/compulim), in the following PRs:
   -  PR [#2540](https://github.com/microsoft/BotFramework-WebChat/pull/2540): `useActivities`, `useReferenceGrammarID`, `useSendBoxDictationStarted`
   -  PR [#2541](https://github.com/microsoft/BotFramework-WebChat/pull/2541): `useStyleOptions`, `useStyleSet`
   -  PR [#2542](https://github.com/microsoft/BotFramework-WebChat/pull/2542): `useLanguage`, `useLocalize`, `useLocalizeDate`
   -  PR [#2543](https://github.com/microsoft/BotFramework-WebChat/pull/2543): `useAdaptiveCardsHostConfig`, `useAdaptiveCardsPackage`, `useRenderMarkdownAsHTML`
   -  PR [#2544](https://github.com/microsoft/BotFramework-WebChat/pull/2544): `useAvatarForBot`, `useAvatarForUser`
   -  PR [#2547](https://github.com/microsoft/BotFramework-WebChat/pull/2547): `useEmitTypingIndicator`, `usePeformCardAction`, `usePostActivity`, `useSendEvent`, `useSendFiles`, `useSendMessage`, `useSendMessageBack`, `useSendPostBack`
   -  PR [#2548](https://github.com/microsoft/BotFramework-WebChat/pull/2548): `useDisabled`
   -  PR [#2549](https://github.com/microsoft/BotFramework-WebChat/pull/2549): `useSuggestedActions`
   -  PR [#2550](https://github.com/microsoft/BotFramework-WebChat/pull/2550): `useConnectivityStatus`, `useGroupTimestamp`, `useTimeoutForSend`, `useUserID`, `useUsername`
   -  PR [#2551](https://github.com/microsoft/BotFramework-WebChat/pull/2551): `useLastTypingAt`, `useSendTypingIndicator`, `useTypingIndicator`
   -  PR [#2552](https://github.com/microsoft/BotFramework-WebChat/pull/2552): `useFocusSendBox`, `useScrollToEnd`, `useSendBoxValue`, `useSubmitSendBox`, `useTextBoxSubmit`, `useTextBoxValue`
   -  PR [#2553](https://github.com/microsoft/BotFramework-WebChat/pull/2553): `useDictateInterims`, `useDictateState`, `useGrammars`, `useMarkActivityAsSpoken`, `useMicrophoneButton`, `useShouldSpeakIncomingActivity`, `useStartDictate`, `useStopDictate`, `useVoiceSelector`, `useWebSpeechPonyfill`

## Description

Add several post message related hooks:

- `useDictateInterims`
- `useDictateState`
- `useGrammars`
- `useMarkActivityAsSpoken`
- `useMicrophoneButton`
- `useShouldSpeakIncomingActivity`
- `useStartDictate`
- `useStopDictate`
- `useVoiceSelector`
- `useWebSpeechPonyfill`

## Specific Changes

- Added the following hooks:
   - `useDictateInterims`
   - `useDictateState`
   - `useGrammars`
   - `useMarkActivityAsSpoken`
   - `useMicrophoneButton`
   - `useShouldSpeakIncomingActivity`
   - `useStartDictate`
   - `useStopDictate`
   - `useVoiceSelector`
   - `useWebSpeechPonyfill`
- Updated components to use aforementioned hooks

---

-  [x] Testing Added

### CODE REVIEW

File: packages/component/src/Dictation.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ const Dictation = ({ emitTypingIndicator, onError }) => {
         if (transcript) {
           setSendBox(transcript);
           submitSendBox('speech');
-          setShouldSpeakIncomingActivity();
+          setShouldSpeakIncomingActivity(true);
```
Comment (tdurnford) [inline packages/component/src/Dictation.js]: Why is this change necessary?

Comment (compulim) [reply packages/component/src/Dictation.js]: It was called `startSpeakActivity` and `stopSpeakActivity`. And now `shouldSpeakIncomingActivity`. We should always call it with `true`/`false`.

And I found test failed when I forget to set it to `true`.
